packages/terra-tabs/src/common-tabs/_Tabs.jsx

```diff
@@ -59,22 +67,52 @@ const propTypes = {
    */
   onChange: PropTypes.func,
   /**
-   * Whether or not the tab is draggable.
+   * Callback function when add button selection has changed.
+   * Parameters: 1. Event 2. Selected pane's key
    */
+  onSelectAddButton: PropTypes.func,
+  /**
+   * Callback function when a tab is closing.
+   * It receives two parameters: 1. Item key of the tab to close. 2. MetaData associated with the tab.
+   */
+  onClosingTab: PropTypes.func,
+
+  /**
+   * Indicates whether the tabs are closable, allowing the user to close tabs.
+   */
+  isClosable: PropTypes.bool.isRequired,
+
+  /**
+   * Callback function when a tab is closed.
+   * It receives two parameters: 1. Item key of the closed tab. 2. MetaData associated with the closed tab.
+   */
+  onClose: PropTypes.func,
+
+  /**
+   * Callback function to handle changes in the tab state.
+   * It receives the updated tab data after a tab is closed.
+   */
+  onTabStateChange: PropTypes.func,
+  /**
+  * Whether or not the tab is draggable.
+  */
   isDraggable: PropTypes.bool,
   /**
   * Callback function triggered when tab is drag and dropped .
-   */
+  */
   onTabOrderChange: PropTypes.func,
 };
 
+let addTabId;
+let addButtonToggle = false;
```

Comment:
These variables should not be declared outside the class. Declare it inside and use this keyword.

Reply:
Done https://github.com/cerner/terra-framework/commit/eba857452cdb936a451cc229d605f0db4d53a97c